### PR TITLE
automated push

### PR DESCRIPTION
**Pull request by romanie**
Moving MAP error into function.

### CODE REVIEW

File: diplom.py [hunk 1]
```diff
@@ -9,6 +9,20 @@ def parse(line):
     label = int (tokens[23])
     return (features, label)
 
+def Mean_Average_Precision (Predicted_probability_distribution, n_of_classes, list_of_classes, test_y, TEST_DATASET_SIZE):
```
Comment (romanie) [inline diplom.py]: Rename:
Mean_Average_Precision -> mean_average_precision
Predicted_probability_distribution -> predicted_probability_distribution

Comment (romanie) [inline diplom.py]: Also put arguments in separate lines:
def func_name(a,
                         b,
                         c):

Comment (romanie) [inline diplom.py]: Rename TEST_DATASET_SIZE -> dataset_size

File: diplom.py [hunk 2]
```diff
@@ -9,6 +9,20 @@ def parse(line):
...
+    AP = 0
+    k = -1
+    for row in Predicted_probability_distribution:
+        k += 1
+        r =[]
```
Comment (romanie) [inline diplom.py]: "r =[]" => "r = []"

Comment (romanie) [inline diplom.py]: rename r to classes_by_prob

File: diplom.py [hunk 3]
```diff
@@ -9,6 +9,20 @@ def parse(line):
...
+        for i in xrange(0, n_of_classes):
+            r.append([row[i], list_of_classes[i]])
+            r.sort(reverse=True)    
+        for j in range (0, 5):
+            if r[j][1] == test_y[k]:
+                AP += 1.0/(j+1)
```
Comment (romanie) [inline diplom.py]: add space around "/"

File: diplom.py [hunk 4]
```diff
@@ -41,7 +55,8 @@ def parse(line):
 print('LogisticRegression score: %f'
       % logistic.fit(train_X, train_y).score(test_X, test_y))
 Predicted_probability_distribution = logistic.predict_proba(test_X)
-#print logistic.classes_
+n_of_classes = len(logistic.classes_)
+list_of_classes = logistic.classes_
 
 AP = 0
 k = -1
```
Comment (romanie) [inline diplom.py]: Add a function call instead of inline code block.

File: diplom.py [hunk 5]
```diff
@@ -9,8 +9,27 @@ def parse(line):
...
+            classes_by_prob.sort(reverse=True)    
+        for j in range (0, 5):
+            if classes_by_prob[j][1] == test_y[k]:
+                AP += 1.0 / (j+1)
+    MAP = AP / dataset_size
+    return MAP
+
 TRAIN_DATASET_SIZE = 30000
-TEST_DATASET_SIZE = 30000
+dataset_size = 30000
```
Comment (romanie) [inline diplom.py]: Rename this constant back to TEST_DATASET_SIZE. I've asked you to only rename argument of the function ;)

File: diplom.py [hunk 6]
```diff
@@ -40,44 +59,24 @@ def parse(line):
...
-    r =[]
-    for i in xrange(0, len(logistic.classes_)):
-        r.append([row[i], logistic.classes_[i]])
-    r.sort(reverse=True)    
-    for j in range (0, 5):
-        if r[j][1] == test_y[k]:
-            AP += 1.0/(j+1)
-MAP = AP / TEST_DATASET_SIZE
-print (MAP)
+print MAP
```
Comment (romanie) [inline diplom.py]: Please output error on train and test datasets. This way you would be able to check basic expectation that your algorithm performs better on the data it already saw than on the new data. BTW, you can respond to my comments here.

Comment (oksanaie) [inline diplom.py]: done, check, please

File: diplom.py [hunk 7]
```diff
@@ -9,6 +9,25 @@ def parse(line):
     label = int (tokens[23])
     return (features, label)
 
+def mean_average_precision (predicted_probability_distribution, 
+    n_of_classes,
```
Comment (romanie) [inline diplom.py]: Do you need n_of_classes here? I think you can derive it from other arguments i.e. len(list_of_classes). Please also add an assertion i.e. "assert len(list_of_classes) == len(predicted_probability_distribution)".

Comment (romanie) [inline diplom.py]: Also, do you need dataset_size? I think you can derive it from the predicted_probability_distribution i.e. len(predicted_probability_distribution).

File: diplom.py [hunk 8]
```diff
@@ -9,6 +9,25 @@ def parse(line):
...
+    k = -1
+    for row in predicted_probability_distribution:
+        k += 1
+        classes_by_prob = []
+        for i in xrange(0, n_of_classes):
+            classes_by_prob.append([row[i], list_of_classes[i]])
+            classes_by_prob.sort(reverse=True)    
+        for j in range (0, 5):
+            if classes_by_prob[j][1] == test_y[k]:
+                AP += 1.0 / (j+1)
```
Comment (romanie) [inline diplom.py]: Add spaces around "+".

File: diplom.py [hunk 9]
```diff
@@ -9,6 +9,25 @@ def parse(line):
     label = int (tokens[23])
     return (features, label)
 
+def mean_average_precision (predicted_probability_distribution, 
+    n_of_classes, 
+    list_of_classes, 
+    test_y, 
+    dataset_size):
+    AP = 0
```
Comment (romanie) [inline diplom.py]: Rename AP to total_error

File: diplom.py (hunk 7 above)
Comment (oksanaie) [inline diplom.py]: Please also add an assertion i.e. "assert len(list_of_classes) == len(predicted_probability_distribution)".
len(list_of_classes) = 100
len(predicted_probability_distribution) = TEST_DATASET_SIZE

File: diplom.py [hunk 10]
```diff
@@ -9,6 +9,23 @@ def parse(line):
...
+    for row in predicted_probability_distribution:
+        k += 1
+        classes_by_prob = []
+        for i in xrange(0, len(list_of_classes)):
+            classes_by_prob.append([row[i], list_of_classes[i]])
+            classes_by_prob.sort(reverse=True)    
+        for j in range (0, 5):
+            if classes_by_prob[j][1] == test_y[k]:
+                total_error += 1.0 / (j + 1)
+    MAP = tot / len(predicted_probability_distribution)
```
Comment (romanie) [inline diplom.py]: You don't need MAP. Just return total_error / len(...)

File: diplom.py [hunk 11]
```diff
@@ -9,6 +9,23 @@ def parse(line):
     label = int (tokens[23])
     return (features, label)
 
+def mean_average_precision (predicted_probability_distribution,
```
Comment (romanie) [inline diplom.py]: align arguments to parenthesis

Comment (oksanaie) [inline diplom.py]: I don't see difference

File: diplom.py [hunk 12]
```diff
@@ -9,6 +9,23 @@ def parse(line):
...
+    for row in predicted_probability_distribution:
+        k += 1
+        classes_by_prob = []
+        for i in xrange(0, len(list_of_classes)):
+            classes_by_prob.append([row[i], list_of_classes[i]])
+            classes_by_prob.sort(reverse=True)    
+        for j in range (0, 5):
+            if classes_by_prob[j][1] == test_y[k]:
+                total_error += 1.0 / (j + 1)
+    MAP = total_error / len(predicted_probability_distribution)
```
Comment (romanie) [inline diplom.py]: You don't need 'MAP', just return the result.

File: diplom.py [hunk 13]
```diff
@@ -40,44 +57,26 @@ def parse(line):
...
-    for i in xrange(0, len(logistic.classes_)):
-        r.append([row[i], logistic.classes_[i]])
-    r.sort(reverse=True)    
-    for j in range (0, 5):
-        if r[j][1] == test_y[k]:
-            AP += 1.0/(j+1)
-MAP = AP / TEST_DATASET_SIZE
-print (MAP)
+MAP_test = mean_average_precision(predicted_probability_distribution_test, 
+    list_of_classes,
```
Comment (romanie) [inline diplom.py]: You can use same practice of aligning arguments here - it improves code readability.

File: diplom.py [hunk 14]
```diff
@@ -40,44 +57,26 @@ def parse(line):
     multi_class = 'multinomial', max_iter = 100)
 print('LogisticRegression score: %f'
       % logistic.fit(train_X, train_y).score(test_X, test_y))
-Predicted_probability_distribution = logistic.predict_proba(test_X)
-#print logistic.classes_
+predicted_probability_distribution_test = logistic.predict_proba(test_X)
+predicted_probability_distribution_train = logistic.predict_proba(train_X)
+list_of_classes = logistic.classes_
```
Comment (romanie) [inline diplom.py]: you don't need this local variable just inline it.

Comment (oksanaie) [inline diplom.py]: which one?

Comment (romanie) [inline diplom.py]: The one you define on the line that I am commenting ;) list_of_classes

Comment (oksanaie) [inline diplom.py]: I didn't understand how to use this platform

Comment (oksanaie) [inline diplom.py]: and what means "inline"?

Comment (romanie) [inline diplom.py]: Inline in this case means remove the variable definition and just pass "logistic.classes_" as argument two function calls.

**Pull request merged**